src/cowboy_swagger.erl

```diff
@@ -120,3 +124,35 @@ normalize_list_values(List) ->
         [V | Acc]
       end,
   lists:foldl(F, [], List).
+
+%% @private
+validate_swagger_map([], Acc) ->
+  Acc;
+validate_swagger_map([K | T], Acc) ->
+  case maps:get(K, Acc, undefined) of
+    undefined ->
+      validate_swagger_map(T, Acc);
+    Val ->
+      Params = validate_swagger_map_params(maps:get(parameters, Val, []), []),
+      Responses = maps:from_list(validate_swagger_map_responses(
+        maps:to_list(maps:get(responses, Val, #{})), [])),
+      NewVal = Val#{parameters => Params, responses => Responses},
+      validate_swagger_map(T, maps:put(K, NewVal, Acc))
+  end.
+
+%% @private
+validate_swagger_map_params([], Acc) ->
+  Acc;
+validate_swagger_map_params([Param | T] = Params, Acc) ->
+  DefName = binary:list_to_bin(
+    [<<"param">>, integer_to_binary(length(Params))]),
```

Comment:
And this is also kind of inefficient and harder to read. You can just build the binary in one step.

``` erlang
<<"param", (integer_to_binary(length(Params)))/binary>>
```